src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1164,7 +1173,12 @@ Node* StoreVectorMaskedNode::Ideal(PhaseGVN* phase, bool can_reshape) {
     if (ty && ty->is_con()) {
       BasicType mask_bt = Matcher::vector_element_basic_type(in(4));
       int load_sz = type2aelembytes(mask_bt) * ty->get_con();
-      assert(load_sz <= MaxVectorSize, "Unexpected store size");
+      if (load_sz > MaxVectorSize) {
+        // See LoadVectorMaskedNode::Ideal
```

Comment:
```suggestion
        // After loop opts, cast nodes are aggressively removed, if the input is then transformed
        // into a constant that is outside the range of the removed cast, we may encounter it here.
        // This should be a dead node then.
```
Optional: Might as well just repeat the explanation. If the code in `LoadVectorMaskedNode::Ideal` changes it is unlikely that we would notice here, and then we'd have a dead link.